pubsublite/internal/wire/subscriber.go

```diff
@@ -0,0 +1,321 @@
+	// Immutable after creation.
+	subClient    *vkit.SubscriberClient
+	settings     ReceiveSettings
+	subscription subscriptionPartition
+	initialReq   *pb.SubscribeRequest
+	receiver     MessageReceiverFunc
+
+	// Fields below must be guarded with mutex.
+	stream          *retryableStream
+	acks            *ackTracker
+	offsetTracker   subscriberOffsetTracker
+	flowControl     flowControlBatcher
+	pollFlowControl *periodicTask
+	seekInFlight    bool
+
+	abstractService
+}
+
+func newSubscribeStream(ctx context.Context, subClient *vkit.SubscriberClient, settings ReceiveSettings,
+	receiver MessageReceiverFunc, subscription subscriptionPartition, acks *ackTracker, disableTasks bool) *subscribeStream {
+
+	s := &subscribeStream{
+		subClient:    subClient,
+		settings:     settings,
+		subscription: subscription,
+		initialReq: &pb.SubscribeRequest{
+			Request: &pb.SubscribeRequest_Initial{
+				Initial: &pb.InitialSubscribeRequest{
+					Subscription: subscription.Path,
+					Partition:    int64(subscription.Partition),
+				},
+			},
+		},
+		receiver: receiver,
+		acks:     acks,
+	}
+	s.stream = newRetryableStream(ctx, s, settings.Timeout, reflect.TypeOf(pb.SubscribeResponse{}))
+
+	backgroundTask := s.sendBatchFlowControl
+	if disableTasks {
+		backgroundTask = func() {}
+	}
+	s.pollFlowControl = newPeriodicTask(batchFlowControlPeriod, backgroundTask)
+	return s
+}
+
+// Start establishes a subscribe stream connection and initializes flow control
+// tokens from ReceiveSettings.
+func (s *subscribeStream) Start() {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+
+	if s.unsafeUpdateStatus(serviceStarting, nil) {
+		s.stream.Start()
+		s.pollFlowControl.Start()
+
+		s.flowControl.OnClientFlow(flowControlTokens{
+			Bytes:    int64(s.settings.MaxOutstandingBytes),
+			Messages: int64(s.settings.MaxOutstandingMessages),
+		})
+	}
+}
+
+// Stop immediately terminates the subscribe stream.
+func (s *subscribeStream) Stop() {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	s.unsafeInitiateShutdown(serviceTerminating, nil)
+}
+
+func (s *subscribeStream) newStream(ctx context.Context) (grpc.ClientStream, error) {
+	return s.subClient.Subscribe(addSubscriptionRoutingMetadata(ctx, s.subscription))
+}
+
+func (s *subscribeStream) initialRequest() (interface{}, bool) {
+	return s.initialReq, true
+}
+
+func (s *subscribeStream) validateInitialResponse(response interface{}) error {
+	subscribeResponse, _ := response.(*pb.SubscribeResponse)
+	if subscribeResponse.GetInitial() == nil {
+		return errInvalidInitialSubscribeResponse
+	}
+	return nil
+}
+
+func (s *subscribeStream) onStreamStatusChange(status streamStatus) {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+
+	switch status {
+	case streamConnected:
+		s.unsafeUpdateStatus(serviceActive, nil)
+
+		// Reinitialize the offset and flow control tokens when a new subscribe
+		// stream instance is connected.
+		if seekReq := s.offsetTracker.RequestForRestart(); seekReq != nil {
+			if s.stream.Send(&pb.SubscribeRequest{
```

Comment:
Nit: do you mind adding a comment explaining why it's safe to ignore the return value of Send()?

Reply:
Done. Thanks for reviewing!